node/src/types/appendable_block.rs

```diff
@@ -220,57 +337,107 @@ impl AppendableBlock {
         self.timestamp
     }
 
+    fn category_count(&self, category: &TransactionV1Category) -> usize {
+        self.transactions
+            .iter()
+            .filter(|(_t, c)| c == category)
+            .count()
+    }
+
+    fn total_approval_count(&self) -> usize {
+        self.transactions
+            .iter()
+            .map(|(t, _c)| t.approvals().len())
+            .sum()
+    }
+
+    fn category_approval_count(&self, category: &TransactionV1Category) -> usize {
+        self.transactions
+            .iter()
+            .filter(|(_t, c)| c == category)
+            .map(|(t, _c)| t.approvals().len())
+            .sum()
+    }
+
     /// Returns `true` if the number of transfers is already the maximum allowed count, i.e. no
     /// more transfers can be added to this block.
     fn has_max_transfer_count(&self) -> bool {
-        self.transfers.len() == self.transaction_config.block_max_transfer_count as usize
+        self.category_count(&TransactionV1Category::Transfer)
+            == self.transaction_config.block_max_transfer_count as usize
     }
 
-    /// Returns `true` if the number of deploys is already the maximum allowed count, i.e. no more
-    /// deploys can be added to this block.
-    fn has_max_deploy_count(&self) -> bool {
-        self.deploys.len() == self.transaction_config.block_max_standard_count as usize
+    /// Returns `true` if the number of transactions is already the maximum allowed count, i.e. no
+    /// more transactions can be added to this block.
+    fn has_max_standard_count(&self) -> bool {
+        self.category_count(&TransactionV1Category::Standard)
+            == self.transaction_config.block_max_standard_count as usize
     }
 
-    /// Returns `true` if adding the deploy with 'additional_approvals` approvals would exceed the
-    /// approval limits.
-    /// Note that we also disallow adding deploys with a number of approvals that would make it
-    /// impossible to fill the rest of the block with deploys that have one approval each.
+    /// Returns `true` if adding the transaction with 'additional_approvals` approvals would exceed
```

Comment:
Smallest nitpick in the world:
```suggestion
    /// Returns `true` if adding the transaction with `additional_approvals` approvals would exceed
```

Reply:
Nice one :+1: 
Fixed in a7df733516eca5a1655c6d094e05b20898c5a8c2